lib/3scale_toolbox/cli.rb

```diff
@@ -12,9 +14,137 @@ def self.load_builtin_commands
   end
 
   def self.run(args)
-    load_builtin_commands
-    ThreeScaleToolbox.load_plugins
-    # TODO handle error
-    root_command.build_command.run args
+    error_watchdog do
+      load_builtin_commands
+      ThreeScaleToolbox.load_plugins
+      root_command.build_command.run args
+    end
+  end
+
+  def self.error_watchdog(exit_on_error: true)
+    new.error_watchdog(exit_on_error: exit_on_error) { yield }
+  end
+
+  # Catches errors and prints nice diagnostic messages
+  def error_watchdog(exit_on_error:)
+    # Set exit handler
+    %w[INT TERM].each do |signal|
+      Signal.trap(signal) do
+        puts
+        exit!(0)
+      end
+    end
+
+    # Set stack trace dump handler
+    if !defined?(RUBY_ENGINE) || RUBY_ENGINE != 'jruby'
+      Signal.trap('USR1') do
+        puts 'Caught USR1; dumping a stack trace'
+        puts caller.map { |i| "  #{i}" }.join("\n")
+      end
+    end
+
+    # Run
+    yield
+  rescue StandardError, ScriptError => e
+    handle_error(e, exit_on_error: exit_on_error)
+  end
+
+  private
+
+  def handle_error(error, exit_on_error:)
+    if expected_error?(error)
+      warn
+      warn "\e[1m\e[31mError: #{error.message}\e[0m"
+    else
+      print_error(error)
+    end
+    exit(1) if exit_on_error
```

Comment:
This is passing a parameter 3 levels deep just to call exit. IMO this should be done in `self.run`. That is what I meant by doing it in the CLI module. Not moving the whole method, but just this action.